### ISSUE TITLE
project settings ignored when building with CodeLite Make Generator

### ISSUE THREAD

**Issue by moliad** [eranif/codelite#2593]
Codelite v14.0.0, Running on windows 8.1, mingw64 (may be related to #2340)

Trying to build an exe, using a workspace with two projects, one is an exe, the other is a static lib.  The compilation of the lib itself is ok, I actually get a lib file on disk and no errors or warnings on compilation and linking.

problem is the intermediate and working directory setups are completely ignored.   Since I'm using linked files (Add existing file)  the .o files are generated in extremely obfuscated paths, always under build-Debug and the lib is always under /lib folder.

for example, here is the -o argument given to gcc for one of my files: 
`-o ../../../../cable/build/codelite/build-Debug/__/__/__/__/common-c-libs/build/codelite/common-c-libs/up_up_up_src_clibs-selftest.c.o`

but with the config setup, I am expecting:
`T:/tmp/selftest.c.o` 
furthermore, if I change the OutputFile path and name, it is named properly but codelite ignores the path information.

Here is a pic of my settings.

![codelite-prefs](https://user-images.githubusercontent.com/475008/96082272-eb3e4580-0e88-11eb-9c64-a29b422541f0.jpg)

I have been testing out all the GCC compatible IDEs (codeblocks, vscode, eclipse, etc,) and so I far Codelite is the only one I feel has the right balance of project setup and simplicity.  I really hope this is a simple thing to solve or incomprehension on my part.

**Comment by moliad** [comment]
Doing a bit more investigation, I've come to understand (I think) the way the workspace, projects and builds interact.  

When I change my project configs and build, I see it updates the .mk file within my project's folder.  That's where the issues seems to arise.  The .mk file doesn't follow the information in the project settings GUI.

whatever I put in the GUI, the projets .mk  IntermediateDirectory and Outdir Variable consistently stay the same.

**Comment by eranif** [comment]
Have you tried using the "Default" build system? it should behave as expected. The `CodeLite Makefile Generator` (the name was renamed in 14.0.4) *always* create a folder at the root workspace named `build-<Configuration Name>` and places the objects in it. The obfuscated paths yo are seeing is the relative path to the workspace location of the object file. The method of keeping the path to the object name is used to allow multiple objects with the same name to co-exist (this is because multiple files can share the file name (e.g. `utils.cpp`) but at a different location. Using your method, you will end up with a single object file `utils.cpp.o` while you actually should have 2 (or more) this will cause a link errors

To summarize:
Change the build system to  `Default`

**Comment by Eric-01** [comment]
Eranif, what about changing the old default (still called "Default" in various places) to "Legacy" or something similar (to reduce confusion)?

**Comment by Eric-01** [comment]
@eranif, what about changing the old default (still called "Default" in various places) to "Legacy" or something similar (to reduce confusion)?
